crates/egui/src/memory.rs

```diff
@@ -68,23 +71,37 @@ pub struct Memory {
     // ------------------------------------------
     /// new scale that will be applied at the start of the next frame
     #[cfg_attr(feature = "persistence", serde(skip))]
-    pub(crate) new_pixels_per_point: Option<f32>,
+    pub(crate) override_pixels_per_point: Option<f32>,
 
     /// new fonts that will be applied at the start of the next frame
     #[cfg_attr(feature = "persistence", serde(skip))]
     pub(crate) new_font_definitions: Option<epaint::text::FontDefinitions>,
 
+    #[cfg_attr(feature = "persistence", serde(skip))]
+    pub(crate) interactions: ViewportIdMap<Interaction>,
+
     #[cfg_attr(feature = "persistence", serde(skip))]
     pub(crate) interaction: Interaction,
 
+    // Current viewport
+    #[cfg_attr(feature = "persistence", serde(skip))]
+    pub(crate) viewport_id: ViewportId,
+
     #[cfg_attr(feature = "persistence", serde(skip))]
     pub(crate) window_interaction: Option<window::WindowInteraction>,
 
+    #[cfg_attr(feature = "persistence", serde(skip))]
+    pub(crate) window_interactions: ViewportIdMap<window::WindowInteraction>,
+
     #[cfg_attr(feature = "persistence", serde(skip))]
     pub(crate) drag_value: crate::widgets::drag_value::MonoState,
 
+    #[cfg_attr(feature = "persistence", serde(skip))]
     pub(crate) areas: Areas,
 
+    #[cfg_attr(feature = "persistence", serde(skip))]
+    pub(crate) viewports_areas: ViewportIdMap<Areas>,
```

Comment:
Same here - I don't think we need to duplicate `Interaction`, `WindowInteraction` and `Areas`. We can just use `interactions.entry(viewport_id)` instead of `interaction`, etc.

Reply:
I tried to do that!
But requires to many changes!
And will be worst performance because needs a lookup every time!

I am thinking to move all of that to a `ViewportMemory` or something.

What is your opinion?

Reply:
I'll fix